elastic/kibana
# [Stateful sidenav] Welcome tour

In this PR I've added the "Solution view welcome tour".

The tour only shows in the following condition

* When there is only 1 space (the "default") and it has a `solution` set
* If it hasn't been closed by the user

Other rules in place:
* Once the first user closes the tour, other users won't see the tour (the flag is stored in Global settings)
* If the users edits the default space and sets a solution, the tour won't show, even if there is only one space
* If the user has 2 spaces and deletes 1 space, the tour won't show (even if the space has a solution set)
* When the user sees the tour, if they click on the space drop down to open it and then they click on the "Manage spaces" button, the tour is hidden and not shown anymore. We assume that they took action from the tour message.

Fixes https://github.com/elastic/kibana-team/issues/1142

## Screenshots

<img width="681" alt="Screenshot 2024-10-08 at 13 29 43" src="https://github.com/user-attachments/assets/0e3f9b35-7e77-47ff-ba17-10d57dee816a">


## How to test

* Add the following in your `kibana.dev.yml`

```yml
xpack.cloud.id: "ftr_fake_cloud_id:aGVsbG8uY29tOjQ0MyRFUzEyM2FiYyRrYm4xMjNhYmM="
xpack.cloud.base_url: "https://cloud.elastic.co"
```

* Launch Kibana and navigate to `/app/management/kibana/spaces/edit/default`
* Edit the default space by setting one of the solution view (e.g. "search")
* Delete the Global settings es doc sending this request

```curl
curl -XDELETE -u system_indices_superuser:changeme  "http://localhost:9200/.kibana/_doc/config-global:9.0.0" -H "kbn-xsrf: spaces"
```
* Refresh the browser window
* You should see the tour dialog
